interface/src/instruction.rs

```diff
@@ -0,0 +1,1697 @@
+use serde_derive::{Deserialize, Serialize};
+use solana_instruction::{AccountMeta, Instruction};
+use solana_pubkey::Pubkey;
+
+use crate::{program::ID, NONCE_STATE_SIZE, RECENT_BLOCKHASHES_ID, RENT_ID};
+
+/// An instruction to the system program.
+#[cfg_attr(
+    feature = "frozen-abi",
+    solana_frozen_abi_macro::frozen_abi(digest = "2LnVTnJg7LxB1FawNZLoQEY8yiYx3MT3paTdx4s5kAXU"),
+    derive(
+        solana_frozen_abi_macro::AbiExample,
+        solana_frozen_abi_macro::AbiEnumVisitor
+    )
+)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
```

Comment:
I added a `serde` feature but it is enable by default since the `new_with_bincode` used in all instructions expects a `data: &T` parameter that is T: serde::Serialize`.

Reply:
In the end I removed the `serde` feature, so it is included by default. `cargo hack check` was not happy about it since it is a "required" dependency.